lib/heroku/command/pg.rb

```diff
@@ -588,21 +588,13 @@ def links
 
   private
 
-  def resolve_service_or_url(name_or_url, default=nil)
-    if name_or_url =~ %r{(postgres://|redis://)}
-      url = name_or_url
-      uri = URI.parse(url)
-      name = url_name(uri)
-      MaybeAttachment.new(name, url, nil)
-    else
-      attachment_name = name_or_url || default
-      attachment = (resolve_addon(attachment_name) || []).first
+  def resolve_service(name)
+    attachment = (resolve_addon(name) || []).first
 
-      error("Remote database could not be found.") unless attachment
-      error("Remote database is invalid.") unless attachment['addon_service']['name'] =~ /heroku-(redis|postgresql)/
+    error("Remote database could not be found.") unless attachment
+    error("Remote database is invalid.") unless attachment['addon_service']['name'] =~ /heroku-(redis|postgresql)/
 
-      MaybeAttachment.new(attachment_name, get_config_var(attachment['config_vars'].first), attachment)
-    end
+    MaybeAttachment.new(attachment['name'], nil, attachment)
```

Comment:
`resolve_addon` doesn't return an attachment, though, it returns an add-on resource. I'm confused what this is doing. Is there code here assuming that this will be an attachment?